### DISCUSSION TITLE
Playing Youtube video with MediaElement

### DISCUSSION THREAD

**Discussion by snice1981** [CommunityToolkit/Maui#2361] [Q&A]
Hi

I'm trying to play youtube video using MediaElement. It was working for some videos but now I always get "Please sign in" message

<img width="155" alt="image" src="https://github.com/user-attachments/assets/2501332c-7948-4161-8ea0-2c31dd768ff8">

I precise that the videos I tried have no restrictions (age or something else)

Does anyone have this problem?

**Comment by bijington** [comment]
I am guessing YouTube expects you to be signed in, we don't currently support passing in things like bearer tokens to the request but I think it is on the list

**Reply by snice1981** [reply]
Thanks for your swift reply.

So does it mean that no one is using it to watch YT videos?

**Reply by bijington** [reply]
I suspect so as I tested it and observed the same thing. I am not sure if we have a proposal to customise what information is sent in the headers but I think that is what will be required

**Reply by ne0rrmatrix** [reply]
I have YouTube working but it requires the correct url. Supporting that I think we need to not support unless we use YouTube API. I did not use API but a nuget package that would never be allowed to be in a store app.

**Reply by snice1981** [reply]
Not sure to correctly understand your answer... but could give more details? What is the correct url?

**Reply by GeorgeLeithead** [reply]
> I have YouTube working but it requires the correct url. Supporting that I think we need to not support unless we use YouTube API. I did not use API but a nuget package that would never be allowed to be in a store app.

A reply made by @ne0rrmatrix on our Discord server on this, is to have a look at this YT repository, which "should help you get the youtube url for using in media element".

[YoutubeExplode](https://github.com/Tyrrrz/YoutubeExplode)
